asset/src/_kafka_clients/producer-client.ts

```diff
@@ -87,9 +101,13 @@ export default class ProducerClient extends BaseClient<kafka.Producer> {
                 // flush the messages at the end of each slice
                 if (i === endOfBatchIndex) {
                     await this._try(() => this._flush(), 'produce', 0);
+                    this._logger.debug(`End of message batch Flushing queue before writing more messages...`);
```

Comment:
I'd move this message above the `this._flush()` so it's consistent with the branch below, and we see the message even if `_flush()` doesn't return.